cmd/rpcdaemon/main.go

```diff
@@ -50,7 +52,22 @@ func main() {
 		defer db.Close()
 		defer engine.Close()
 
-		apiList := jsonrpc.APIList(db, backend, txPool, mining, ff, stateCache, blockReader, cfg, engine, logger, nil)
+		polygonSync, err := cmd.PersistentFlags().GetBool("polygon.sync")
+		if err != nil {
+			return err
+		}
+
+		var bridgeReader bridge.ReaderService
+		if polygonSync {
+			stateReceiverContractAddress := params.BorMainnetChainConfig.Bor.GetStateReceiverContract() // mainnet and amoy share the same address
```

Comment:
params.BorMainnetChainConfig

what if it is running Amoy?
can't just hardcode and assume they are the same on Amoy and Maine

Reply:
Updated to accept this as a param with the default as the address for Amoy and Mainnet.